src/components/Clock/Clock.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+
+type State = {
+  time: string;
+  intervalId?: NodeJS.Timeout | number;
+};
+
+type Props = {
+  clockName: number | string;
+};
+
+export class Clock extends React.Component<Props, State> {
+  state = {
+    time: new Date().toLocaleTimeString(),
+    intervalId: 0,
+  };
+
+  componentDidMount() {
+    const intervalId = setInterval(() => {
+      this.setState({ time: new Date().toLocaleTimeString() });
+    }, 1000);
+
+    this.setState({ intervalId });
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const oldName = prevProps.clockName;
+
+    return oldName !== this.props.clockName
+    // eslint-disable-next-line
```

Comment:
disable eslint for console.log for the entire file. The code would be much readable in this case

Reply:
🔥